src/main/java/io/xcherry/core/persistence/Persistence.java

```diff
@@ -95,13 +148,31 @@ public List<AppDatabaseRow> getAppDatabaseRows(final String tableName) {
         return appDatabase.get(tableName);
     }
 
-    public AppDatabaseWrite getAppDatabaseWrite(final DatabaseStringEncoder encoder) {
+    public Object getLocalAttribute(final String key) {
+        if (!localAttributes.containsKey(key)) {
+            throw new LocalAttributeNotFoundException(String.format("Key %s does not exist in local attributes", key));
+        }
+
+        return localAttributes.get(key);
+    }
+
+    /**
+     * Update or insert value for a local attribute.
+     *
+     * @param key       key of the local attribute.
+     * @param value     value of the local attribute.
+     */
+    public void upsertLocalAttribute(final String key, final Object value) {
```

Comment:
probably just `setLocalAttribute` to pair with `getLocalAttribute`. You can keep the comment to make it more clear